File: src/main/java/ordilov/randomplay/playlist/domain/PlaylistItem.java
package ordilov.randomplay.playlist.domain;

import static javax.persistence.FetchType.LAZY;
import static javax.persistence.GenerationType.IDENTITY;
import static lombok.AccessLevel.PROTECTED;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import lombok.Getter;
import lombok.NoArgsConstructor;
import ordilov.randomplay.track.domain.Track;

@Entity
@Getter
@NoArgsConstructor(access = PROTECTED)
public class PlaylistItem {

  @Id
  @GeneratedValue(strategy = IDENTITY)
  private Long id;

  @ManyToOne(fetch = LAZY)
  @JoinColumn(name = "playlist_id")
  private Playlist playlist;

  @ManyToOne(fetch = LAZY)
  @JoinColumn(name = "track_id")
  private Track track;

  public PlaylistItem(Playlist playlist, Track track) {
    this.playlist = playlist;
    this.track = track;
  }
}

File: src/main/java/ordilov/lissn/playlist/domain/PlaylistItem.kt
package ordilov.lissn.playlist.domain;

import ordilov.lissn.common.domain.BaseEntity
import ordilov.lissn.track.domain.Track
import javax.persistence.*

@Entity
data class PlaylistItem(
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "playlist_id")
    val playlist: Playlist,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "track_id")
    val track: Track,

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null
)